pkg/app/server.go

```diff
@@ -99,9 +101,25 @@ func RunKubeStateMetrics(ctx context.Context, opts *options.Options, factories .
 			ConstLabels: prometheus.Labels{"handler": "metrics"},
 		}, []string{"method"},
 	)
+	configHash := promauto.With(ksmMetricsRegistry).NewGauge(
+		prometheus.GaugeOpts{
+			Name: "kube_state_metrics_config_hash",
+			Help: "Hash of the currently loaded configuration.",
+		})
+	configSuccess := promauto.With(ksmMetricsRegistry).NewGauge(
+		prometheus.GaugeOpts{
+			Name: "kube_state_metrics_config_last_reload_successful",
```

Comment:
I'd prefer `kube_state_metrics_last_config_reload_successful`

Reply:
I tend to agree, in future we might have other config files as well though, that's why I added the infix.
Another config we might want to create metrics for once the hot reload for it is in place as well:
`kube_state_metrics_customresourcestateconfig_last_reload_successful`

Alternatively we could drop it from the name and use:

```
kube_state_metrics_last_reload_successful{type="config"}
kube_state_metrics_last_reload_successful{type="customresourcestateconfig"}
```

Reply:
I'd vote for using labels, seems to reduce verbosity and users can also monitor all config updates through a single metric.